test/common/http/codec_client_test.cc

```diff
@@ -229,7 +229,10 @@ TEST_F(CodecClientTest, IdleTimerClientLocalCloseWithActiveRequests) {
 }
 
 TEST_F(CodecClientTest, ProtocolError) {
-  EXPECT_CALL(*codec_, dispatch(_)).WillOnce(Throw(CodecProtocolException("protocol error")));
+  EXPECT_CALL(*codec_, dispatch(_)).WillOnce(Invoke([](Buffer::Instance&) -> absl::Status {
+    throw CodecProtocolException("protocol error");
+    return absl::OkStatus();
```

Comment:
Is this actually required? I would think the compiler would realize this can't be reached? Same elsewhere?

Reply:
Thanks! It looks like it would complain if the expectation had the function signature -> void  like below, so I did change those to `WillOnce(Throw(_))` instead of the Invokes